src/sentry/eventstore/snuba_discover/backend.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import
+
+import six
+from copy import deepcopy
+
+from sentry.utils import snuba
+from sentry.eventstore.base import EventStorage
+from sentry.eventstore.snuba.backend import (
+    ASC_ORDERING,
+    DESC_ORDERING,
+    get_after_event_condition,
+    get_before_event_condition,
+    SnubaEventStorage,
+)
+
+
+class SnubaDiscoverEventStorage(EventStorage):
+    """
+    Experimental backend that uses the Snuba Discover dataset instead of Events
+    or Transactions directly.
+    """
+
+    def get_events(self, *args, **kwargs):
+        return SnubaEventStorage().get_events(*args, **kwargs)
```

Comment:
Discover will need this method long term too. Are you thinking that the discover implementation will be used by eventstore long term, or will be have another service implementation for interacting with discover dataset?

Reply:
Yeah I think the goal is for all methods of Eventstore to return the underlying transactions or events as appropriate. I'll add an implementation for this and `get_event_by_id` that uses the Discover dataset as well.

Reply:
What prevents us from using the discover dataset for this and get_event_by_id now ?

Reply:
No real blockers, just a bit of extra work to get these methods working since they return SnubaEvent and use dataset_query which does a bunch of other things like map the column names.

Reply:
So you plan to eventually move the whole backend to discover, is that correct ?

Reply:
Yes